lib/chef/azure/commands/enable.rb

```diff
@@ -112,31 +113,38 @@ def configure_chef_only_once
 
       load_settings
 
-      # Write validation key
-      File.open("#{bootstrap_directory}/validation.pem", "w") do |f|
-        f.write(@validation_key)
-      end
-
-      # Write client.rb
-      File.open("#{bootstrap_directory}/client.rb", "w") do |f|
-        f.write(override_clientrb_file(@client_rb))
-      end
-      # write the first_boot.json
-      File.open("#{bootstrap_directory}/first-boot.json", "w") do |f|
-        f.write(<<-RUNLIST
-{
-"run_list": [#{ @run_list.empty? ? "" : escape_runlist(@run_list)}]
-}
-RUNLIST
-)
-      end
-
       # run chef-client for first time with no runlist to register the node
       puts "Running chef client for first time with no runlist..."
 
       begin
-        params = " -c #{bootstrap_directory}/client.rb -E _default -L #{@azure_plugin_log_location}/chef-client.log --once "
-        result = shell_out("chef-client #{params}")
+        require 'chef/azure/core/bootstrap_context'
+        config = {}
+        config[:environment] = value_from_json_file(handler_settings_file,'runtimeSettings','0','handlerSettings', 'publicSettings', 'bootstrap_options','environment')
+        config[:chef_node_name] = value_from_json_file(handler_settings_file,'runtimeSettings','0','handlerSettings', 'publicSettings', 'bootstrap_options','chef_node_name')
+        config[:encrypted_data_bag_secret ] = value_from_json_file(handler_settings_file,'runtimeSettings','0','handlerSettings', 'publicSettings', 'bootstrap_options','encrypted_data_bag_secret')
+        config[:chef_extension_root] = @chef_extension_root
+        Chef::Config[:validation_key_content] = @validation_key
+        Chef::Config[:chef_server_url] = value_from_json_file(handler_settings_file,'runtimeSettings','0','handlerSettings', 'publicSettings', 'bootstrap_options','chef_server_url')
+        Chef::Config[:validation_client_name] = value_from_json_file(handler_settings_file,'runtimeSettings','0','handlerSettings', 'publicSettings', 'bootstrap_options','validation_client_name')
+
+        if windows?
+          context = Chef::Knife::Core::WindowsBootstrapContext.new(config, {}, Chef::Config)
+          template_file = Gem.find_files(File.join("chef","azure","bootstrap","windows-chef-client-msi.erb")).first

```

Comment:
Is it a right way to find file path? I think Gem.find_files used to find files in other gems?